src/aggregators/aggregators.jl

```diff
@@ -184,6 +184,32 @@ needs_vartojumps_map(aggregator::RSSACR) = true
 supports_variablerates(aggregator::AbstractAggregatorAlgorithm) = false
 supports_variablerates(aggregator::Coevolve) = true
 
+# true if aggregator supports hops, e.g. diffusion
 is_spatial(aggregator::AbstractAggregatorAlgorithm) = false
 is_spatial(aggregator::NSM) = true
 is_spatial(aggregator::DirectCRDirect) = true
+
+# return the fastest aggregator out of the available ones
+function select_aggregator(jumps::JumpSet; vartojumps_map=nothing, jumptovars_map=nothing, dep_graph=nothing, spatial_system=nothing, hopping_constants=nothing)
+    
+    # detect if a spatial SSA should be used
+    !isnothing(spatial_system) && !isnothing(hopping_constants) && return DirectCRDirect
+
+    # if variable rate jumps are present, return the only SSA that supports them
+    num_vrjs(jumps) != 0 && return Coevolve
+    
+    # if the number of jumps is small, return the Direct
+    num_majumps(jumps) + num_crjs(jumps) < 10 && return Direct
+
+    # if there are only massaction jumps, we can any build dependency graph, so return the fastest aggregator
+    num_crjs(jumps) == 0 && num_vrjs(jumps) == 0 && return RSSACR
+
+    # in the presence of constant rate jumps, we rely on the given dependency graphs
+    if !isnothing(vartojumps_map) && !isnothing(jumptovars_map)
+        return RSSACR
+    elseif !isnothing(dep_graph)
+        return DirectCR
+    else
+        return Direct
```

Comment:
I think we are missing a middle regime where `SortingDirect` or `RSSA` might be better. Have you played around with this at all / looked over the SciMLBenchmarks and the Catalyst paper results?

Reply:
When I did the Catalyst tests, `RSSA` was best for small systems. However, we didn't test very small systems, so not sure about these. But there definitely was a regime where `RSSA` as the best choice.

Reply:
![image](https://github.com/SciML/JumpProcesses.jl/assets/17322558/d20d0e54-72fb-4d37-951d-8f875773a4e2)
This image suggests that RSSA is indeed better on small models but only marginally. I did not see a comparison with Direct and SortingDirect there. @TorkelE , do you have a table of times from when you did the benchmarking? I did not find such a table for SSA benchmarks in the paper.

Reply:
![image](https://github.com/SciML/JumpProcesses.jl/assets/17322558/41837f53-bee8-46ac-b19a-f786b9c044ff)
![image](https://github.com/SciML/JumpProcesses.jl/assets/17322558/cdb17dc8-3574-4457-9315-320995849c21)
These two are from models with 6 and 18 reactions, respectively. Indeed, it looks like between ~6 and ~66 reactions we might want to use RSSA. The number 66 comes from the benchmark from the Catalyst paper.

Reply:
Think the smallest model (Multistate) is 18 reactions. It is kind of hard to see the Direct and Sorting DIrect as RSSA (and RSSACR) is more or less on top of them (also suggesting that the choice of method is not that important).

There is not good table, but the results are in JSON files here: https://github.com/SciML/Catalyst_PLOS_COMPBIO_2023/tree/master/Benchmarks/Benchmarking_results/Threads_1

Look for e.g. RSSA_multistate and you find the multistate results. The data is saved in `Dict`s with the benchmark median times and simulation lengs as separate entries.

Reply:
Does that hold with static array states?

Reply:
I added `RSSA` if the number of jumps is between 10 and 100. This schema is bound to be just a heuristic so I don't think we should spend a ton of time finding the exact threshold when e.g. RSSA is better than e.g. SortingDirect.